File: tests/e2e/fixtures/unsigned_integers/mysql/java/src/main/java/com/example/db/com/example/db/queries/Queries.java
package com.example.db.queries;

import com.example.db.models.UnsignedValues;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public final class Queries {
    private Queries() {}

    private static final String SQL_INSERT_UNSIGNED_ROW = """
            INSERT INTO unsigned_values (u8_val, u16_val, u24_val, u32_val, u64_val)
            VALUES (?, ?, ?, ?, ?);
            """;
    public static void insertUnsignedRow(Connection conn, short u8Val, int u16Val, long u24Val, long u32Val, java.math.BigInteger u64Val) throws SQLException {
        try (PreparedStatement ps = conn.prepareStatement(SQL_INSERT_UNSIGNED_ROW)) {
            ps.setShort(1, u8Val);
            ps.setInt(2, u16Val);
            ps.setLong(3, u24Val);
            ps.setLong(4, u32Val);
            ps.setBigDecimal(5, new java.math.BigDecimal(u64Val));
            ps.executeUpdate();
        }
    }

    private static final String SQL_GET_UNSIGNED_ROWS = """
            SELECT id, u8_val, u16_val, u24_val, u32_val, u64_val
            FROM unsigned_values
            ORDER BY id;
            """;
    public static List<UnsignedValues> getUnsignedRows(Connection conn) throws SQLException {
        try (PreparedStatement ps = conn.prepareStatement(SQL_GET_UNSIGNED_ROWS)) {
            List<UnsignedValues> rows = new ArrayList<>();
            try (ResultSet rs = ps.executeQuery()) {
                while (rs.next()) rows.add(new UnsignedValues(rs.getObject(1, java.math.BigInteger.class), rs.getShort(2), rs.getInt(3), rs.getLong(4), rs.getLong(5), rs.getObject(6, java.math.BigInteger.class)));
            }
            return rows;
        }
    }

    private static Boolean getNullableBoolean(java.sql.ResultSet rs, int col) throws java.sql.SQLException {
        boolean v = rs.getBoolean(col);
        return rs.wasNull() ? null : v;
    }
    private static Short getNullableShort(java.sql.ResultSet rs, int col) throws java.sql.SQLException {
        short v = rs.getShort(col);
        return rs.wasNull() ? null : v;
    }
    private static Integer getNullableInt(java.sql.ResultSet rs, int col) throws java.sql.SQLException {
        int v = rs.getInt(col);
        return rs.wasNull() ? null : v;
    }
    private static Long getNullableLong(java.sql.ResultSet rs, int col) throws java.sql.SQLException {
        long v = rs.getLong(col);
        return rs.wasNull() ? null : v;
    }
    private static Float getNullableFloat(java.sql.ResultSet rs, int col) throws java.sql.SQLException {
        float v = rs.getFloat(col);
        return rs.wasNull() ? null : v;
    }
    private static Double getNullableDouble(java.sql.ResultSet rs, int col) throws java.sql.SQLException {
        double v = rs.getDouble(col);
        return rs.wasNull() ? null : v;
    }
}

File: tests/e2e/fixtures/unsigned_integers/mysql/kotlin/src/main/kotlin/com/example/db/com/example/db/queries/Queries.kt
package com.example.db.queries

import java.sql.Connection
import com.example.db.models.UnsignedValues

object Queries {

    private val SQL_INSERT_UNSIGNED_ROW = """
        INSERT INTO unsigned_values (u8_val, u16_val, u24_val, u32_val, u64_val)
        VALUES (?, ?, ?, ?, ?);
    """.trimIndent()
    fun insertUnsignedRow(conn: Connection, u8Val: Short, u16Val: Int, u24Val: Long, u32Val: Long, u64Val: java.math.BigInteger): Unit {
        conn.prepareStatement(SQL_INSERT_UNSIGNED_ROW).use { ps ->
            ps.setShort(1, u8Val)
            ps.setInt(2, u16Val)
            ps.setLong(3, u24Val)
            ps.setLong(4, u32Val)
            ps.setBigDecimal(5, java.math.BigDecimal(u64Val))
            ps.executeUpdate()
        }
    }

    private val SQL_GET_UNSIGNED_ROWS = """
        SELECT id, u8_val, u16_val, u24_val, u32_val, u64_val
        FROM unsigned_values
        ORDER BY id;
    """.trimIndent()
    fun getUnsignedRows(conn: Connection): List<UnsignedValues> {
        conn.prepareStatement(SQL_GET_UNSIGNED_ROWS).use { ps ->
            val rows = mutableListOf<UnsignedValues>()
            ps.executeQuery().use { rs ->
                while (rs.next()) rows.add(UnsignedValues(rs.getObject(1, java.math.BigInteger::class.java), rs.getShort(2), rs.getInt(3), rs.getLong(4), rs.getLong(5), rs.getObject(6, java.math.BigInteger::class.java)))
            }
            return rows
        }
    }

    private fun getNullableBoolean(rs: java.sql.ResultSet, col: Int): Boolean? {
        val v = rs.getBoolean(col)
        return if (rs.wasNull()) null else v
    }
    private fun getNullableShort(rs: java.sql.ResultSet, col: Int): Short? {
        val v = rs.getShort(col)
        return if (rs.wasNull()) null else v
    }
    private fun getNullableInt(rs: java.sql.ResultSet, col: Int): Int? {
        val v = rs.getInt(col)
        return if (rs.wasNull()) null else v
    }
    private fun getNullableLong(rs: java.sql.ResultSet, col: Int): Long? {
        val v = rs.getLong(col)
        return if (rs.wasNull()) null else v
    }
    private fun getNullableFloat(rs: java.sql.ResultSet, col: Int): Float? {
        val v = rs.getFloat(col)
        return if (rs.wasNull()) null else v
    }
    private fun getNullableDouble(rs: java.sql.ResultSet, col: Int): Double? {
        val v = rs.getDouble(col)
        return if (rs.wasNull()) null else v
    }

    @Suppress("UNCHECKED_CAST")
    private fun <T> jdbcArrayToList(arr: java.sql.Array): List<T> =
        (arr.array as Array<T>).toList()
}
